build/CodeSign.props

```diff
@@ -0,0 +1,274 @@
+    <FilesToSign Include="StackExchange.Redis.StrongName.dll" Certificate="$(AssemblySigning3rdPartyCertName)" Container="Microsoft.AspNetCore.All" />
+    <FilesToSign Include="System.Interactive.Async.dll" Certificate="$(AssemblySigning3rdPartyCertName)" Container="Microsoft.AspNetCore.App" />
+
+    <!-- These files should already be signed already by a different leg of the build. They have to be listed again here because we recreate a redistributable which binaries built in other repos. -->
+    <FilesToExcludeFromSigning Include="libuv.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.AI.DependencyCollector.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.ApplicationInsights.AspNetCore.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.ApplicationInsights.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.Azure.KeyVault.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.Azure.KeyVault.WebKey.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.Azure.Services.AppAuthentication.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.Data.Edm.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.Data.Edm.resources.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.Data.OData.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.Data.OData.resources.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.Extensions.DiagnosticAdapter.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.Extensions.ObjectPool.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.IdentityModel.Clients.ActiveDirectory.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.IdentityModel.Clients.ActiveDirectory.Platform.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.IdentityModel.JsonWebTokens.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.IdentityModel.Logging.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.IdentityModel.Protocols.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.IdentityModel.Protocols.OpenIdConnect.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.IdentityModel.Protocols.WsFederation.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.IdentityModel.Tokens.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.IdentityModel.Tokens.Saml.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.IdentityModel.Xml.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.Rest.ClientRuntime.Azure.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.Rest.ClientRuntime.dll" />
+    <FilesToExcludeFromSigning Include="Microsoft.WindowsAzure.Storage.dll" />
+
+    <!-- These files should already be signed by the .NET Core team. They have to be listed again here because we recreate a redistributable which includes the Microsoft.NETCore.App runtime. -->
+    <FilesToExcludeFromSigning Include="api-ms-win-core-console-l1-1-0.dll" />
```

Comment:
Oh boy, what a list! Is there any way for this to be data-driven somehow? Like we can look at the DLLs, and say "oh we know this came from .NET, and it is indeed already signed, so exclude it"? I can't tell how much of a maintenance nightmare this will be.

Reply:
For managed files, we could use public key tokens. Supposedly, if a managed file has one of .NET team's strong name keys, we can map it to "Microsoft". 

For native files, like api-ms-win-core-console-l1-1-0.dll, there isn't really a good way.

We've also discussed using a fallback like "*.dll", but I'm concerned this might pick up things which it shouldn't, so I'd rather not go there.

Reply:
OK we can revisit in the future if this becomes burdensome.